dapp/src/utils/hooks.js

```diff
@@ -1,57 +1,103 @@
 import { useEffect, useState, useRef } from 'react'
 import { useWeb3React } from '@web3-react/core'
-
-import { injected, connectorsByName, getConnectorImage } from './connectors'
+import { SafeAppConnector } from '@gnosis.pm/safe-apps-web3-react'
+
+import {
+  injected,
+  connectorsByName,
+  getConnector,
+  getConnectorImage,
+} from './connectors'
 import AccountStore from 'stores/AccountStore'
 import analytics from 'utils/analytics'
 
 export function useEagerConnect() {
   const { activate, active } = useWeb3React()
 
-  const [tried, setTried] = useState(false)
+  const [triedInjected, setTriedInjected] = useState(false)
+  const [triedSafeMultisig, setTriedSafeMultisig] = useState(false)
+  const [safeMultisigConnector, setSafeMultisigConnector] = useState(null)
+  const [connector, setConnector] = useState(null)
+  const [connectorIcon, setConnectorIcon] = useState(null)
+
+  // Instantiate Gnosis Safe web3-react connector
+  useEffect(() => {
+    // Requires global.window to be available to must be instantiated inside
+    // useEffect for Next.js to work
+    setSafeMultisigConnector(new SafeAppConnector())
```

Comment:
I wonder if it would be cleaner adding this connector to `utils/connectors` to `connectorsByName`. Each item there would also need a `showOnLoginPanel`, where Gnosis would have that property to `false`. 

The one challenge would be initialisation of the connector since it needs to happen in browser environment.

Reply:
That was my first approach but it broke the next build for the reasons you describe (at least the static export which I was using for IPFS). I think we probably want to keep that available.

Reply:
Aha I see thanks